src/coreclr/src/inc/clrconfigvalues.h

```diff
@@ -395,6 +395,9 @@ RETAIL_CONFIG_DWORD_INFO(INTERNAL_TraceInterpreterIL, W("TraceInterpreterIL"), 0
 RETAIL_CONFIG_DWORD_INFO(INTERNAL_TraceInterpreterOstack, W("TraceInterpreterOstack"), 0, "Logs operand stack after each IL instruction of interpreted methods to the console")
 CONFIG_DWORD_INFO(INTERNAL_TraceInterpreterVerbose, W("TraceInterpreterVerbose"), 0, "Logs interpreter progress with detailed messages to the console")
 CONFIG_DWORD_INFO(INTERNAL_TraceInterpreterJITTransition, W("TraceInterpreterJITTransition"), 0, "Logs when the interpreter determines a method should be JITted")
+RETAIL_CONFIG_DWORD_INFO(INTERNAL_EnableJitOptimization, W("EnableJitOptimization"), 0, "If non-zero, enable JIT optimization (in Debug/Checked)") // this has no effect in Release
```

Comment:
This super confusing. I am not able figure out what this is supposed to do by looking at the code.

Reply:
Should this be rather called `ForceJitOptimization`?

Reply:
Hi @jkotas! Indeed I am a bit confused here. By looking at the code here https://github.com/dotnet/runtime/blob/6d3a9779737f27c33effd5e46c7e2d819b3d7e56/src/coreclr/src/vm/method.cpp#L4868
What name and description do you think is appropriate for this variable?

I initially named this variable as `ForJitOptimization`, but @noahfalk mentioned [here](https://github.com/dotnet/runtime/pull/37825#discussion_r439725270) that it should be instead named to be `AllowJitOptimizationForDebugCode`. By looking at the code, the variable is used in the function `IsJitOptimizationDisabled`. Therefore upon some reflection I am feeling that returning `false` in this function means that we enable JIT optimization. What do you think?

Reply:
Here is how the desired optimization level of a method is determined by default:
- If the binary is compiled as Debug, the optimizations are on
- If the binary is compiled as Release (or Checked), the optimizations are on.

This per binary default can be further over-riden by debugger or by per method `[MethodImpl(MethodImplOptions.NoOptimization)]` attribute.

If I am reading the code correctly, the switch as you have it implemented says ignore all this and set the desired optimization level to be optimizations on, not matter what other parts of the system say.

It means that turning in on may even produce incorrect behavior in certain situations. Respecting `NoOptimization` is often required for functional correctness. I expect that you may see some test failures if you turn this on in its current form.

Reply:
fyi I commented on another thread of discussion - I'm still not sure this config switch is needed at all. If it isn't needed then it will shortcut discussions about getting the policy and naming clear : ) If it is needed then hopefully a better understanding of the requirements will make it easier to get the right policy and right name.